### ISSUE TITLE
bpf: Argument list too long. Program  too large

### ISSUE THREAD

**Issue by thx1111** [mechpen/sockdump#12]
Arch Linux
linux 5.17.9.arch1-1
bcc 0.24.0-2
python-bcc 0.24.0-2
clang 14.0.6-1
libbpf 0.8.0-1

```
$ sudo ./sockdump.py --format string /run/opendkim/opendkim.sock
bpf: Argument list too long. Program  too large (1155 insns), at most 4096 insns

Traceback (most recent call last):
  File "/local/./sockdump.py", line 348, in <module>
    main(args)
  File "/local/./sockdump.py", line 292, in main
    b.attach_kprobe(
  File "/usr/lib/python3.10/site-packages/bcc/__init__.py", line 836, in attach_kprobe
    fn = self.load_func(fn_name, BPF.KPROBE)
  File "/usr/lib/python3.10/site-packages/bcc/__init__.py", line 522, in load_func
    raise Exception("Failed to load BPF program %s: %s" %
Exception: Failed to load BPF program b'probe_unix_socket_sendmsg': Argument list too long
```

https://stackoverflow.com/questions/70147464/program-too-large-threshold-greater-than-actual-instruction-count
"Program too large" threshold greater than actual instruction count

>On Linux 5.15 with a privileged user, the verifier gives up after reading 1 million instructions (the complexity limit). Since it has to analyze all paths through the program, a program with a small number of instructions can have a very high complexity.
...
Since the bpf(2) syscall returns E2BIG both when the program is too large and when its complexity is too high, libbpf prints the same error message for both cases, always with at most 4096 instructions.

The bpf Verifier is a moving target...

**Comment by mechpen** [comment]
This should be fixed in 9d87426c517b. Please verify.

**Comment by thx1111** [comment]
Ha!  Yes, thank you.  Works for me.
